R/learner_randomPlantedForest_classif_rpf.R

```diff
@@ -0,0 +1,130 @@
+        split_try = p_int(lower = 1, upper = Inf, default = 10, tags = "train"),
+        t_try = p_dbl(lower = 0, upper = 1, default = 0.4, tags = "train"),
+        loss = p_fct(c("L1", "L2", "logit", "exponential"), default = "exponential", tags = "train"),
+        delta = p_dbl(lower = 0, upper = 1, default = 1, tags = "train"),
+        epsilon = p_dbl(lower = 0, upper = 1, default = 0.1, tags = "train"),
+        deterministic = p_lgl(default = FALSE, tags = "train"),
+        nthreads = p_int(lower = 1, upper = Inf, default = 1, tags = c("train", "threads")),
+        cv = p_lgl(default = FALSE, tags = "train"),
+        purify = p_lgl(default = FALSE, tags = "train")
+      )
+
+      param_set$values = list(loss = "exponential", max_interaction_limit = 30)
+
+      super$initialize(
+        id = "classif.rpf",
+        packages = "randomPlantedForest",
+        feature_types = c("integer", "numeric", "factor", "ordered"),
+        predict_types = c("response", "prob"),
+        param_set = param_set,
+        properties = c("twoclass", "multiclass"),
+        man = "mlr3extralearners::mlr_learners_classif.rpf",
+        label = "Random Planted Forest"
+      )
+    }
+  ),
+  private = list(
+    .train = function(task) {
+      # get parameters for training
+      pars = self$param_set$get_values(tags = "train")
+      # max_interaction_limit is needed but must not be passed to rpf(),
+      # while convert_ratio automatically removes max_interaction_ratio.
+      max_interaction_limit = pars[["max_interaction_limit"]]
+      pars[["max_interaction_limit"]] = NULL
+      n_features = length(task$feature_names)
+
+      pars = convert_ratio(
+        pars, "max_interaction", "max_interaction_ratio",
+        min(n_features, max_interaction_limit)
+      )
+
+      invoke(
+        randomPlantedForest::rpf,
+        x = task$data(cols = task$feature_names),
+        y = task$data(cols = task$target_names),
+        .args = pars
+      )
+    },
+    .predict = function(task) {
+      # get parameters with tag "predict"
+      pars = self$param_set$get_values(tags = "predict")
+
+      # get newdata and ensure same ordering in train and predict
+      newdata = task$data(cols = names(self$state$data_prototype))
+
+      # Calculate predictions for the selected predict type.
+      # type = self$predict_type
+
+      if (self$predict_type == "response") {
+        pred = invoke(
+          predict, self$model, new_data = newdata,
+          type = "class", .args = pars
+        )
+        list(response = pred[[".pred_class"]])
+      } else {
+        pred = invoke(
+          predict, self$model, new_data = newdata,
+          type = "prob", .args = pars
+        )
+        # Result will be a df with one column per variable with names '.pred_<level>'
+        # we want the names without ".pred"
+        names(pred) = sub(pattern = ".pred_", replacement = "", names(pred))
```

Comment:
what happens here if a column is called `a.pred_` ?

Reply:
Ah, fair point, that was a somewhat lazy hack. I tried to make it more robust, but it doesn't feel like a particularly elegant solution.

Reply:
Perfect, but please use  `=` instead of `<-`